src/java.desktop/share/classes/javax/swing/SwingWorker.java

```diff
@@ -552,6 +550,7 @@ public final void execute() {
      * {@inheritDoc}
      */
     public final boolean cancel(boolean mayInterruptIfRunning) {
+        setState(StateValue.DONE);
         return future.cancel(mayInterruptIfRunning);
```

Comment:
Setting the state to `DONE` unconditionally doesn't seem right either. What if `doInBackground` isn't started? What if `doInBackground` starts after you compare the state is `PENDING`?

If `cancel` changes the state, it must do it after `future.cancel` returns. But then, the state moves to `DONE` before `doInBackground` exited which contradicts the spec for `DONE`.

Reply:
OK..I have removed unconditional setting of DONE state..
But then as I mentioned, it would cause JCK failure, so I have reverted isDone() change too...

Reply:
> OK..I have removed unconditional setting of DONE state.. But then as I mentioned, it would cause JCK failure, so I have reverted isDone() change too...

Should we update the spec for `isDone()`? I think it's the right way because it makes the behaviour consistent. Doing so requires a CSR which will be the grounds for updating the JCK test.

Reply:
I am not sure if it is entirely wrong..I will keep it as it is for now..I dont want to go in CSR route for now..
The present iteration takes care of reproducer and also satisifies JCK..
If in future it is needed, we can take a look..

Reply:
Let's see what others have to say.